src/OpenTelemetry.Instrumentation.Http/HttpRequestMessageContextPropagation.cs

```diff
@@ -34,6 +34,11 @@ internal static class HttpRequestMessageContextPropagation
 
         internal static Action<HttpRequestMessage, string, string> HeaderValueSetter => (request, name, value) =>
         {
+            if (request.Headers.Contains(name))
+            {
+                request.Headers.Remove(name);
+            }
```

Comment:
```suggestion
request.Headers.Remove(name);
```

[Remove](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.remove?view=net-6.0#system-net-http-headers-httpheaders-remove(system-string)) already handles it not being present in the `request.Headers`

Reply:
Thanks!